### PR TITLE
Bug: Fixes where the director name comes from for DACT-1696

### PR DESCRIPTION
**Pull request by cdrainCH**

### CODE REVIEW

File: src/controllers/shared.controllers/director.correspondence.address.manual.controller.ts [hunk 1]
```diff
@@ -171,12 +173,13 @@ export const buildValidationErrors = (validationStatusResponse: ValidationStatus
   return validationErrors;
 }
 
-export const renderPage = (req: Request, res: Response, serviceAddress: Address, officerFiling: OfficerFiling, validationErrors: ValidationError[], templateName: string, backUrlPath: string) => {
+export const renderPage = async (req: Request, res: Response, serviceAddress: Address, officerFiling: OfficerFiling, validationErrors: ValidationError[], templateName: string, backUrlPath: string, isUpdate: boolean, session: Session) => {
```
Comment (mdesaikns) [inline src/controllers/shared.controllers/director.correspondence.address.manual.controller.ts:176]: I'm not quite sure why the sonar has not kicked off build but there's going to be a code smell for number of method params. 
I think you can create a new interface like [renderPageParam](https://github.com/companieshouse/officer-filing-web/blob/main/src/utils/renderArrayPageParams.ts) for your classes to get around it.

File: test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts [hunk 2]
```diff
@@ -87,22 +90,75 @@ const invalidData = {
   "correspondence_address_postcode": "ゃ"
 }
 
+const nullValidationErrors = [
```
Comment (mdesaikns) [inline test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts:93]: do we need these? I am sure there's a way to mock these.

File: src/controllers/shared.controllers/director.correspondence.address.manual.controller.ts (hunk 1 above)
Comment (cdrainCH) [reply src/controllers/shared.controllers/director.correspondence.address.manual.controller.ts:176]: Ah I didn't realise there would a smell for it, will address this now

File: test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts [hunk 3]
```diff
@@ -214,13 +275,15 @@ describe("Director correspondence address controller tests", () => {
...
+          mockValidateManualAddress.mockReturnValue(invalidCharacterValidationErrors);
+          mockGetValidationStatus.mockResolvedValue(mockValidValidationStatusResponse);
           mockGetOfficerFiling.mockResolvedValueOnce({
             data: {
               firstName: "John",
-              lastName: "Smith"
+              lastName: "Smith",
             }
           });
+          mockGetCompanyAppointmentFullRecord.mockResolvedValue({});
```
[the remark this answers is not in the document]
Comment (cdrainCH) [reply test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts:286]: I've tried this but it doesn't seem to work for these tests. Will try again to make sure

File: test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts (hunk 2 above)
Comment (cdrainCH) [reply test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts:93]: I was unsure of how to mock these, so in the interest of saving time I created these to use instead

File: test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdrainCH) [reply test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts]: Mocked properly

File: test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts (hunk 2 above)
Comment (cdrainCH) [reply test/controllers/shared.controllers/director.correspondence.address.manual.controller.unit.ts]: Mocked properly

**Review by mdesaikns — APPROVED** [review]

**Review by clewis1 — APPROVED** [review]

**Pull request merged**